packages/sdk-codegen/src/python.gen.ts

```diff
@@ -276,17 +310,25 @@ ${this.hooks.join('\n')}
    * @attr.s(kw_only=True) we'll only allow kw_args.
    */
   construct(indent: string, type: IType) {
-    // Skip read-only parameters
-    if (!this.methodInputModelTypes.has(type)) return ''
+    if (type instanceof EnumType) return ''
     indent = this.bumper(indent)
     const bump = this.bumper(indent)
-    let result = `\n\n${indent}def __init__(self, *${this.argDelimiter}`
+    const annotations: string[] = []
     const args: string[] = []
     const inits: string[] = []
-    Object.values(type.properties).forEach((prop) => {
-      args.push(this.declareConstructorArg('', prop))
-      inits.push(this.initArg(bump, prop))
-    })
+    this.typeProperties(type)
+      .forEach((prop) => {
+        annotations.push(this.declareProperty(indent, prop, true))
+        args.push(this.declareConstructorArg('', prop))
+        inits.push(this.initArg(bump, prop))
+      })
+    let result = ''
+    if (Object.values(type.properties).some(this.isBareForwardRef)) {
+      result = `\n${indent}__annotations__ = {\n`
+      result += `${annotations.join(',\n')}`
+      result += `\n${indent}}`
+    }
```

Comment:
prepending the `__annotations__` declaration to the constructor - feels a little out of place as it's more in the `declareType` section but I didn't want to override/copy that method in here to modify it.

Reply:
fwiw this could be
```ts
      result = `\n${indent}__annotations__ = {\n` +
        `${annotations.join(',\n')}` +
        `\n${indent}}`
```
but you probably know that

Reply:
oh right, I find that my brain get's a little overloaded when dealing with intricacies of one language and then switching to another - and I forget simpler/stylistic stuff like this :-)